cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/executor/ReplicationThrottleHelper.java

```diff
@@ -260,12 +287,16 @@ private boolean removeFollowerThrottledReplicasFromTopic(Properties config, Stri
   }
 
   private void removeThrottledReplicasFromTopic(String topic, Set<String> replicas) {
-    Properties config = _kafkaZkClient.getEntityConfigs(ConfigType.Topic(), topic);
-    boolean removedLeaderThrottle = removeLeaderThrottledReplicasFromTopic(config, topic, replicas);
-    boolean removedFollowerThrottle = removeFollowerThrottledReplicasFromTopic(config, topic, replicas);
+    Properties topicConfigs = _kafkaZkClient.getEntityConfigs(ConfigType.Topic(), topic);
+    if (topicConfigs == null || topicConfigs.isEmpty()) {
```

Comment:
The current implementation of the `getEntityConfigs` method does not return `null`. However, it's not safe to assume it will never return `null`. We can assume it will not return null only if the API clearly claims its behavior.

Reply:
-- Discussed offline.
We will keep the existing logic because API contract leaves the behavior unspecified and the implementation does not return `null`. Hence, we cannot assume that if a future version happens to start returning `null` it would imply that the topic does not exist (i.e. it can return null for some other irrelevant reason). Thus, It is safer to fail if a behavior change causes a `null` response.